runner/app/live/infer.py

```diff
@@ -74,9 +74,20 @@ async def start_control_subscriber(handler: PipelineStreamer, control_url: str):
         segment = await subscriber.next()
         if segment.eos():
             return
-        params = await segment.read()
-        logging.info("Received control message, updating model with params: %s", params)
-        handler.update_params(**json.loads(params))
+
+        try:
+            params = await segment.read()
+            logging.info("Received control message, updating model with params: %s", params)
+            data = json.loads(params)
+        except Exception as e:
+            logging.error(f"Error parsing control message: {e}")
+            continue
+        
+        try:        
+            handler.update_params(data)
+        except Exception as e:
+            logging.error(f"Error updating model with control message: {e}")
```

Comment:
Makes sense, the intention here wasn't to really return the error as much as prevent it from breaking the running pipeline. This was needed because any error or `return` during the While loop would end the control subscriber task loop. This was a separate change I implemented in https://github.com/livepeer/ai-worker/pull/302 and kept here for testing. I'll rebase and remove the change from this PR.

Reply:
Yeah having the try catch is good!